tests/module/test_misc.js

```diff
@@ -0,0 +1,163 @@
+import * as misc from 'misc';
+import * as ckb from 'ckb';
+
+function test_ckb_smt_verify1() {
+    // Test vectors from the Rust example
+    const keyHex =
+        '381dc5391dab099da5e28acd1ad859a051cf18ace804d037f12819c6fbc0e18b';
+    const valueHex =
+        '9158ce9b0e11dd150ba2ae5d55c1db04b1c5986ec626f2e38a93fe8ad0b2923b';
+    const rootHashHex =
+        'ebe0fab376cd802d364eeb44af20c67a74d6183a33928fead163120ef12e6e06';
+    const proofHex =
+        '4c4fff51ff322de8a89fe589987f97220cfcb6820bd798b31a0b56ffea221093d35f909e580b00000000000000000000000000000000000000000000000000000000000000';
+
+    // Update the conversions to use misc.encoding.decodeHex
+    const key = misc.encoding.decodeHex(keyHex);
+    const value = misc.encoding.decodeHex(valueHex);
+    const rootHash = misc.encoding.decodeHex(rootHashHex);
+    const proof = misc.encoding.decodeHex(proofHex);
```

Comment:
I think it's weird to write ".encoding.decodeXXX", maybe it is better to use "misc.hex.encode/decode and misc.base64.encode/decode"